crates/comms/src/livekit_room.rs

```diff
@@ -104,24 +100,36 @@ fn connect_livekit(
     mut commands: Commands,
     mut new_livekits: Query<(Entity, &mut LivekitTransport), Without<LivekitConnection>>,
     player_state: Res<GlobalCrdtState>,
-    mic: Res<LocalAudioSource>,
+    #[cfg(not(target_arch = "wasm32"))] mic: Res<LocalAudioSource>,
 ) {
     for (transport_id, mut new_transport) in new_livekits.iter_mut() {
         debug!("spawn lk connect");
         let remote_address = new_transport.address.to_owned();
         let receiver = new_transport.receiver.take().unwrap();
         let sender = player_state.get_sender();
 
-        let subscription = mic.subscribe();
+        #[cfg(not(target_arch = "wasm32"))]
+        {
+            let subscription = mic.subscribe();
+            std::thread::spawn(move || {
+                livekit_handler(transport_id, remote_address, receiver, sender, subscription)
+            });
+        }
 
-        std::thread::spawn(move || {
-            livekit_handler(transport_id, remote_address, receiver, sender, subscription)
-        });
+        #[cfg(target_arch = "wasm32")]
+        {
+            // For WASM, we directly call the handler which will spawn the async task
+            let receiver = Arc::new(Mutex::new(receiver));
```

Comment:
nit: you don't need the arc<mutex> here, we have it for native because we use a retry loop so we needed to clone it